servicetalk-http-api/src/test/java/io/servicetalk/http/api/DefaultHttpSetCookiesTest.java

```diff
@@ -91,18 +91,18 @@ private static void decodeSecureCookieNames(final HttpHeaders headers) {
 
     @Test
     void decodeDifferentCookieNames() {
-        final HttpHeaders headers = new ReadOnlyHttpHeaders("set-cookie",
```

Comment:
IIUC `ReadOnlyHttpHeaders` is a pkg-private class that is not used for anywhere except tests. Can you please remove it with corresponding tests?

Reply:
Lets consider removal of ReadOnlyHeaders as a followup PR if necessary. The intention was to potentially use it for some gRPC cases but headers now being exposed in filters makes that difficult todo.

Reply:
Oh, didn't see @Scottmitch's comment in time. I already added a commit that removes it. Which way do you prefer?

Reply:
Let's move the last commit into a separate PR for visibility and easier revert if necessary.